core/modules/user/user.module

```diff
@@ -2528,23 +2528,23 @@ function user_role_save($role) {
  *   The name of the role to be deleted.
  */
 function user_role_delete($role_name) {
-  $role = user_role_load($role_name);
-
-  // If this is the administrator role, delete the user_admin_role variable.
-  if ($role_name == config_get('system.core', 'user_admin_role')) {
-    $system_config = config('system.core');
-    $system_config->clear('user_admin_role');
+  // If this is the administrator role, explicitly set user_admin_role attribute
+  // to 0 ("disabled").
+  $system_config = config('system.core');
+  if ($role_name == $system_config->get('user_admin_role')) {
+    $system_config->set('user_admin_role', 0);
     $system_config->save();
   }
 
-  $config = config('user.role.' . $role_name);
-  $config->delete();
+  $user_config = config('user.role.' . $role_name);
+  $user_config->delete();
 
   // Update the users who have this role set:
   db_delete('users_roles')
     ->condition('role', $role_name)
     ->execute();
 
+  $role = user_role_load($role_name);
```

Comment:
This confuses me a bit.
Above you're using the "shortcut" to load the config directly instead of using user_role_load() - `config('user.role.' . $role_name);`

But here you're using user_role_load() again. On the role that has just been deleted. Mind to explain a bit?

Reply:
This is not actually me doing this - I'm just rearranging preexisting code. This line here was initially in line 2531, right after the function declaration.

Reply:
Right, but your rearrangement means that you load the role *after* it has been deleted. Before your change the role has been loaded *before* it got deleted.

This very likely breaks hook_user_role_delete (below).

Reply:
🤔

Reply:
...makes me think if `module_invoke_all()` below should be called earlier (before the role is deleted), or even if we should be passing the role name instead of the role object to the hook.

Anyway, it seems that this change should not be part of this PR. I'm usually good with adding inline comments when making changes that are required, and since I don't see any comments here, I don't think that this was required/intentional.

Reply:
...I've reverted that specific change now.